File: src/api/img.ts
import { invoke } from "@tauri-apps/api";

namespace Img {
    export async function compress(img_path: string, save_path: string, quality: Number) {
        let ret = await invoke<boolean>('img_compress', { imgPath: img_path, savePath: save_path, quality: quality });
        return ret;
    }

    export async function convert(img_path: string, save_path: string, format: "bmp" | "jpg" | "png") {
        let ret = await invoke<boolean>('img_convert', { imgPath: img_path, savePath: save_path, format: format })
        return ret;
    }
}

export default Img;

File: src-tauri/src/api/img.rs
use crate::unit;
#[tauri::command(async)]
pub async fn img_compress(img_path: String, save_path: String, quality: u32) -> bool {
    if img_path.ends_with(".jpg") || img_path.ends_with(".jpeg") {
        let ret = unit::img::compress_jpg(&img_path, &save_path, quality);
        return ret;
    } else if img_path.ends_with(".png") {
        let ret = unit::img::compress_png(&img_path, &save_path, quality);
        return ret;
    } else {
        return false;
    }
}

#[tauri::command(async)]
pub async fn img_convert(img_path: String, save_path: String, format: String) -> bool {
    unit::img::convert(&img_path, &save_path, &format)
}
